lib/plugin.js

```diff
@@ -230,6 +233,21 @@ internals.Plugin.prototype.register = function (plugins /*, [options], callback
             connection._registrations[item.name] = item;
         }
 
+
+        if (item.dependencies.length) {
+
+            // Declare dependencies
+
+            selection.dependency(item.dependencies, function (server, next) {
+
+                // Deferred registration when all dependencies are resolve
```

Comment:
This is a change in behavior. Dependency does not imply order. If you want to dictate order of registration you can't use this mechanism.